### ISSUE TITLE
macro definitions are not scoped properly

### ISSUE THREAD

**Issue by CohenArthur** [Rust-GCC/gccrs#1581]
```rust
extern "C" {
    fn printf(fmt: *const i8, ...);
}

fn print(value: i32) {
    unsafe { printf("%d\n\0" as *const str as *const i8, value) }
}

fn main() {
    macro_rules! a {
        () => {
            14
        };
    }

    {
        macro_rules! a {
            () => {
                15
            };
        }
    }

    print(a!());
}
```

rustc output: 14
gccrs output: 15

**Comment by CohenArthur** [comment]
Needs #1580

**Comment by CohenArthur** [comment]
Fixed by #1750
